### PR TITLE
Parse header if subject includes ! to indicate breaking change

### PR DESCRIPTION
**Pull request by WillGibson**
At the moment `CommitParser.parse` does not set `type`, `scope`, or `subject` when the commit includes an exclamation mark to indicate a breaking change. E.g. `feat(hello/world)!: message`.

This pull request addresses that so that they are available to use post parsing for all commits.

### CODE REVIEW

**Comment by WillGibson** [comment]
What chance is there of a maintainer approving the workflow?

Also, it says "This pull request may be merged without approvals". Is that legit? I would prefer a maintainer to be happy with the change before merging.

**Comment by WillGibson** [comment]
Oh, "Only those with write access to this repository can merge pull requests.". Cool.

**Review by dangreen — APPROVED** [review]

**Pull request merged**
